src/connectors/walletconnect.ts

```diff
@@ -199,10 +214,12 @@ export class WalletConnectConnector extends BaseConnector implements Connector {
           // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
           const address = providerResult.namespaces?.solana?.accounts[0]?.split(':')[2]
 
-          if (address) {
+          if (address && this.qrcode) {
             setAddress(address)
-            console.log({ rs: providerResult })
             resolve(address)
+            importW3mModalCtrl().then(ModalCtrl => {
```

Comment:
There seems to be a chained `.then` flow here, could be flattened out with `await`

Reply:
It's because all this is in a `return new Promise` (to control the flow of `resolve` depending on if `qrcode` is true